controllers/theater.controller.js

```diff
@@ -0,0 +1,234 @@
+        })
+    }
+}
+exports.deleteTheater = async (req, res) => {
+    const id = req.params.id
+    if (!id) {
+        return res.status(400).send({
+            message: "bad request!"
+        })
+    }
+    try {
+        const theater = await THEATER.findOneAndDelete({
+            _id: id
+        })
+        if (!theater) {
+            return res.status(404).send({
+                message: "Theater does not exists for delete.",
+            })
+        }
+        return res.status(200).send({
+            message: "Theater deleted successfully!",
+            deleted_theater: theater
+        })
+    } catch (err) {
+        console.log(err.message)
+        return res.status(500).send({
+            message: "Internal server error!"
+        })
+    }
+}
+
+exports.addMovieToTheater = async (req,res) => {
+    const theaterId = req.params.id
+    const movieIds = req.body.movieId
+    const insert = req.body.insert
+    if (!theaterId || !movieIds) {
+        return res.status(400).send({
+            message: 'bad request!'
+        })
+    }
+    try {
+        const theater = await THEATER.findOne({
+            _id: theaterId
+        })
+        if (insert==true) {
+            movieIds.forEach(movie => {
+                theater.movies.push(movie)
+            });
+        }else {
+            movieIds.forEach(movie => {
+                theater.movies =  theater.movies.filter(id => id!=movie)
+            });
+        }
+        await theater.save()
+        return res.status(200).send({
+            Updated_Theater: theater
+        })
+    } catch (err) {
+        console.log(err.message)
+        return res.status(500).send({
+            message: "Internal server error!"
+        })
+    }
+}
+exports.getTheaterByMovie =  async (req,res)=>{
+    const movieId = req.params.movieId
+    if(!movieId){
+        return res.status(400).send({
+            message: "bad request!"
+        })
+    }
+    try { 
+          const theater = await THEATER.find({movies:movieId})       
+          if(!theater){
+            return res.status(404).send({
+                message: "Theater does not exists!"
+            })
+         } 
+          return res.status(200).send({
+            Theaters : theater
+          })  
+    }catch (err) {
+        console.log(err.message)
+        return res.status(500).send({
+            message: "Internal server error!"
+        })
+    }
+}
+exports.MovieInsideTheTheater = async (req,res)=>{
+    const movieId = req.params.movieId
+    const theaterId = req.params.theaterId
+    if(!movieId || !theaterId){
+        return res.status(400).send({
+            message: "bad request!"
+        })
+    }
+    try {
+         const theater = await THEATER.findOne({_id:theaterId})
+         if(!theater){
+            return res.status(404).send({
+                message: "Theater does not exists!"
+            })
+         }
+         const movie = theater.movies.filter(id => id==movieId)
```

Comment:
try using array.includes method

Reply:
@satagar In this approach we first find the theater and then find its movies. What I am trying to do is directly find the theater which have the movieId using a single find operation, if that's possible. Can you tell me if there is any way I can do that?